packages/ra-core/src/controller/input/useReferenceInputController.ts

```diff
@@ -125,7 +125,10 @@ export const useReferenceInputController = <RecordType extends RaRecord = any>(
         finalData = possibleValuesData;
         finalTotal = total;
     } else {
-        finalData = [referenceRecord, ...possibleValuesData];
+        finalData = [...possibleValuesData];
```

Comment:
This is the only solution I have found for now. 

During a short time, referenceRecord becomes the selected value in the radio group. 
So we have a finalData with a duplicate. 

I will add it in two steps with a check to avoid the rendering jump

Reply:
I suggest you use a JavaScript Set instead

https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/Set

Reply:
I had tried this technique, but it creates a "flick" when rendering the component.


https://user-images.githubusercontent.com/18294269/193303452-018a60de-0dbd-4947-b31e-d44c382c28e0.mp4